test/PublicApiTests/BaseLine/Microsoft.OData.PublicApi.net45.bsl

```diff
@@ -3179,18 +3166,6 @@ public enum Microsoft.OData.Edm.Validation.EdmErrorCode : int {
     InvalidValue = 282
     InvalidVersionNumber = 25
     IsUnboundedCannotBeTrueWhileMaxLengthIsNotNull = 298
-    JsonInvalid = 401
-    JsonInvalidCollectionValue = 410
-    JsonInvalidKeyValue = 411
-    JsonInvalidNumberType = 405
-    JsonInvalidValue = 403
-    JsonInvalidVersionNumber = 402
-    JsonMissingRequiredProperty = 409
-    JsonSchemaCannotHaveMoreThanOneEntityContainer = 412
-    JsonUnexpectedElement = 407
-    JsonUnexpectedValueKind = 408
-    JsonUnknownElementValueKind = 404
-    JsonUnsupportedElement = 406

```

Comment:
Do we need special JSON versions of these? i.e., why have a JsonInvalidValue rather than just InvalidValue?

Reply:
I'd like to specify this error for the JSON related. 
Besides, this is in net45 because the error message txt file doesn't support #if NETSTANDARD ....

---
In reply to: [445639365](https://github.com/OData/odata.net/pull/1826#discussion_r445639365) [](ancestors = 445639365)

Reply:
I don't understand. If I am consuming metadata that may be either XML or JSON (or openApi or some other syntax) why do I need different errors for an invalid value in JSON versus an invalid value in XML or any other format?

---
In reply to: [445690075](https://github.com/OData/odata.net/pull/1826#discussion_r445690075) [](ancestors = 445690075,445639365)

Reply:
Do you mean the "JsonInvalidValue" or all of the Json* error codes?

---
In reply to: [445702146](https://github.com/OData/odata.net/pull/1826#discussion_r445702146) [](ancestors = 445702146,445690075,445639365)

Reply:
Actually, unless it is something that would be JSON in any metadata format then I would use generic errors for all of these.  

---
In reply to: [445713382](https://github.com/OData/odata.net/pull/1826#discussion_r445713382) [](ancestors = 445713382,445702146,445690075,445639365)

Reply:
1) Remove JsonInvalidValue
2)  Remove JsonInvalidVersionNumber, and reuse the InvalidVersionNumver
4)  JsonInvalidKeyValue => InvalidKeyValue
5)  JsonSchemaCannotHaveMoreThanOneEntityContainer =>  SchemaCannotHaveMoreThanOneEntityContainer
6)  JsonInvalidCollectionValue =>InvalidCollectionValue
6)  JsonInvalid =>  InvalidJson
7)  JsonUnknownElementValueKind => UnknownJsonElementValueKind
8)  JsonInvalidNumberType => InvalidJsonNumberType
9)  JsonUnsupportedElement=> UnsupportedJsonElement
10)  JsonUnexpectedElement=> UnexpectedJsonElement
11)  JsonUnexpectedValueKind=> UnexpectedJsonValueKind
12)  JsonMissingRequiredProperty=> MissingRequiredJsonProperty



---
In reply to: [445731299](https://github.com/OData/odata.net/pull/1826#discussion_r445731299) [](ancestors = 445731299,445713382,445702146,445690075,445639365)

Reply:
Added comments to your PR.  I still think a lot of the JSON errors could be general.  For example, is it important that an unexpected element is JSON, or just that an unexpected element was found?  Seems like this is entirely generic, and shouldn't be tied to JSON.  The only one I see an argument for is InvalidJson, and even that could be InvalidPayload.

---
In reply to: [445768953](https://github.com/OData/odata.net/pull/1826#discussion_r445768953) [](ancestors = 445768953,445731299,445713382,445702146,445690075,445639365)